### PR TITLE
AML-2766 Confirm training provider

### PR DESCRIPTION
**Pull request by elliotritchie**

### CODE REVIEW

File: src/SFA.DAS.ProviderRelationships.Database/Scripts/PostDeployment/SeedDevData.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+﻿IF (NOT EXISTS (SELECT 1 FROM [dbo].[Account] WHERE Id = 1))
```
Comment (lazcool) [inline src/SFA.DAS.ProviderRelationships.Database/Scripts/PostDeployment/SeedDevData.sql]: Should be [dbo].[Accounts] - this stops the db publishing!
```suggestion
IF (NOT EXISTS (SELECT 1 FROM [dbo].[Accounts] WHERE Id = 1))
```

File: src/SFA.DAS.ProviderRelationships.Web/Views/Providers/Add.cshtml [hunk 2]
```diff
@@ -19,12 +19,12 @@
...
         {
             @Html.AntiForgeryToken()
-            @Html.HiddenFor(m => m.Ukprn)
+            @Html.HiddenFor(m => m.AddAccountProviderCommand.Ukprn)
             <div class="govuk-form-group @(Html.IsValid(m => m.Choice) ? "" : " govuk-form-group--error")">
                 <fieldset class="govuk-fieldset">
                     <legend class="govuk-fieldset__legend govuk-fieldset__legend--s">
                         <h1 class="govuk-fieldset__heading">
-                            Is this the training provider you would like to set permissions for?
+                            Is this the training provider you would like to add?
```
Comment (lazcool) [inline src/SFA.DAS.ProviderRelationships.Web/Views/Providers/Add.cshtml]: Note, the design attached to the story is out of date, this matches the latest design.

File: src/SFA.DAS.ProviderRelationships.Web/Controllers/ProvidersController.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+            return View(model);
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        [Route("add")]
+        public async Task<ActionResult> Add(AddProviderViewModel model)
+        {
+            var accountProviderId = await _mediator.Send(model.AddAccountProviderCommand);
```
Comment (Pauljgraham) [inline src/SFA.DAS.ProviderRelationships.Web/Controllers/ProvidersController.cs]: shouldn't this add only work if they confirm

File: src/SFA.DAS.ProviderRelationships/Application/Commands/AddAccountProviderCommand.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        [Required]
+        public long? AccountId { get; set; }
+
+        [Required]
+        public Guid? UserRef { get; set; }
+        
+        [Required]
+        public long? Ukprn { get; set; }
+    }
+}
```
Comment (Pauljgraham) [inline src/SFA.DAS.ProviderRelationships/Application/Commands/AddAccountProviderCommand.cs]: We seem to bind our request commands directly to our command handlers. Is this wise? It shouldn't matter here as we control the front end and back end

File: src/SFA.DAS.ProviderRelationships/Application/Queries/SearchProvidersQueryHandler.cs [hunk 5]
```diff
@@ -1,33 +1,36 @@
...
+            if (!isValid)
             {
                 throw new ValidationException().AddError(request, r => r.Ukprn, ErrorMessages.InvalidUkprn);
             }
 
             return new SearchProvidersQueryResponse
             {
-                Ukprn = int.Parse(request.Ukprn)
+                Ukprn = ukprn
             };
```
Comment (Pauljgraham) [inline src/SFA.DAS.ProviderRelationships/Application/Queries/SearchProvidersQueryHandler.cs]: If we are returning a simple type (ie a long) do we really want to wrap it in a class

File: src/SFA.DAS.ProviderRelationships.Web/Controllers/ProvidersController.cs [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        [Route("search")]
+        public ActionResult Search()
+        {
+            return View(new SearchProvidersViewModel());
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        [Route("search")]
+        public async Task<ActionResult> Search(SearchProvidersViewModel model)
```
[the remark this answers is not in the document]
Comment (elliotritchie) [reply src/SFA.DAS.ProviderRelationships.Web/Controllers/ProvidersController.cs]: This has been addressed in PR #30 where the input models only contain properties with primitive types.

File: src/SFA.DAS.ProviderRelationships.Web/Controllers/ProvidersController.cs (hunk 3 above)
Comment (elliotritchie) [reply src/SFA.DAS.ProviderRelationships.Web/Controllers/ProvidersController.cs]: Yes, good spot!

**Comment by elliotritchie** [comment]
Merged into PR #30.

**Pull request closed without merging**